core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/click/ClickModifier.java

```diff
@@ -21,76 +21,48 @@
 import com.cognifide.aet.job.api.collector.CollectorJob;
 import com.cognifide.aet.job.api.exceptions.ParametersException;
 import com.cognifide.aet.job.api.exceptions.ProcessingException;
-import com.cognifide.aet.validation.ValidationResultBuilder;
-
-import org.apache.commons.lang3.StringUtils;
+import com.cognifide.aet.job.common.modifiers.WebElementsLocatorParams;
 import org.openqa.selenium.By;
-import org.openqa.selenium.ElementNotVisibleException;
-import org.openqa.selenium.NoSuchElementException;
-import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
-import java.util.concurrent.TimeUnit;
 
-public class ClickModifier implements CollectorJob {
+public class ClickModifier extends WebElementsLocatorParams implements CollectorJob {
 
   private static final Logger LOG = LoggerFactory.getLogger(ClickModifier.class);
 
   public static final String NAME = "click";
 
   private final WebDriver webDriver;
 
-  private static final String PARAM_XPATH = "xpath";
-
-  private static final String PARAM_TIMEOUT = "timeout";
-
-  private static final long TIMEOUT_SECONDS_MAX_VALUE = 15L;
-
-  private final ValidationResultBuilder validationResultBuilder;
-
-  /**
-   * Xpath to element to click.
-   */
-  private String xpath;
-
-  /**
-   * Timeout for waiting for element (in seconds).
-   */
-  private long timeoutInSeconds;
-
-  public ClickModifier(final WebDriver webDriver, ValidationResultBuilder validationResultBuilder) {
+  public ClickModifier(final WebDriver webDriver) {
     this.webDriver = webDriver;
-    this.validationResultBuilder = validationResultBuilder;
   }
 
-
   @Override
   public CollectorStepResult collect() throws ProcessingException {
     CollectorStepResult result;
     try {
-      WebElement elementToClick = getElementByXpath(webDriver, xpath);
+        By elementLocator = getLocator();
+        waitForElementToBePresent(webDriver, elementLocator);
+        WebElement elementToClick = webDriver.findElement(elementLocator);
+
       if (elementToClick.isDisplayed()) {
         elementToClick.click();
         result = CollectorStepResult.newModifierResult();
       } else {
-        final String message = String.format("Element defined by xpath: '%s' is not yet visible! {}", xpath);
+        final String message = String.format("Element defined by %s: '%s' is not yet visible!", getSelectorType(), getSelectorValue());
```

Comment:
Maybe `By` class has a nice `toString()` method and we could use it instead of saving selector type in `WebElementsLocatorParams` class? (nice to have)